crypto/default/src/main/java/org/keycloak/crypto/hash/Argon2PasswordHashProvider.java

```diff
@@ -98,19 +101,29 @@ public boolean verify(String rawPassword, PasswordCredentialModel credential) {
     }
 
     private String encode(String rawPassword, byte[] salt, String version, String type, int hashLength, int parallelism, int memory, int iterations) {
-        org.bouncycastle.crypto.params.Argon2Parameters parameters = new org.bouncycastle.crypto.params.Argon2Parameters.Builder(Argon2Parameters.getTypeValue(type))
-                .withVersion(Argon2Parameters.getVersionValue(version))
-                .withSalt(salt)
-                .withParallelism(parallelism)
-                .withMemoryAsKB(memory)
-                .withIterations(iterations).build();
-
-        Argon2BytesGenerator generator = new Argon2BytesGenerator();
-        generator.init(parameters);
-
-        byte[] result = new byte[hashLength];
-        generator.generateBytes(rawPassword.toCharArray(), result);
-        return Base64.encodeBytes(result);
+        try {
+            try {
+                cpuCoreSemaphore.acquire();
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
```

Comment:
Isn't the thread already interrupted when an InterruptException is thrown? So is this line needed?

Reply:
An "InterruptedException" is only caught once in a thread. If you don't want to have the next item in the control flow of your thread to also see this interruption as you don't plan to handle the interruption, you'll need to add the flag to the current thread again. 

https://stackoverflow.com/questions/4906799/why-invoke-thread-currentthread-interrupt-in-a-catch-interruptexception-block